go/vt/vttablet/tabletmanager/vreplication/framework_test.go

```diff
@@ -79,13 +80,13 @@ func init() {
 	tabletconntest.SetProtocol("go.vt.vttablet.tabletmanager.vreplication.framework_test", "test")
 
 	binlogplayer.RegisterClientFactory("test", func() binlogplayer.Client { return globalFBC })
-	flag.Set("binlog_player_protocol", "test")
+	pflag.Set("binlog_player_protocol", "test")
```

Comment:
this is racy whether it works or not; see https://github.com/vitessio/vitess/pull/11010 for a deterministic approach to the problem

Reply:
I don't know if I have understood the cause of the raciness correctly. Can you take a look at https://github.com/vitessio/vitess/pull/11095/files#diff-56ead0241f68e27b40866e956a40b8335919c438e888db80cb1c8b2f287ef256R73 and see if the approach is correct? Else please help me understand the issue/solution approach better.

Reply:
`pflag` sets the global flag set, and depends on _someone_ having registered the flag there already (which this test was not doing before), so if that were true, this call would work correctly. if this test happened to be run in a different order (before someone called `registerFlags`), then the call would have no effect and the test would either fail (probably with a weird/confusing error) or, worse, pass even if the change introduced a bug